### DISCUSSION TITLE
How to know all the requests that I left on a website with python?

### DISCUSSION THREAD

**Discussion by Baboomba** [community/community#38365] [Apps, API and Webhooks] [answered]
I am a beginner in the coding world. Now, I'm in trouble writing some codes of a crawling application with python.

I want to know, and catch the path in Response headers. Sure, I can check it with the DevTools of Chrome.
But, it's a little hard to do it with thousands of pages. So, I wonder if there is a way to catch, or hook it. 

The problem is below,

1. url is not changed, if I click any buttons on the website.
2. the parameter(I want to know) can't be checked => the method, parse.urlparse(url) can't be used anymore.
3. Chrome DevTools shows the information(paths, or parameters)
4. But, I don't know how to get it! T.T

**Comment by airtower-luna** [comment]
> I want to know, and catch the path in Response headers. Sure, I can check it with the DevTools of Chrome.

What exactly are you looking for? Usually the requested path (part of the URL) is in the *request*, not the response. Well, unless the response is some kind of redirect. A screenshot of how you can see the thing you're looking for might be helpful to answer this question.

Also, how are you doing the crawling? Which Python modules, function, etc.? Ideally show a code snippet.

**Reply by Baboomba** [reply]
Oh, I made a mistake! I replied your question in the bellow person!

As you can see, I wonder if I can get the 'shopOwnerNumber'.

Because, Chrome DevTools do it, I think there left this information somewhere.

If I can know it, I will get all of our hundreds of stores automatically!

**Comment by david-engelmann** [comment]
depending on what library you are using to crawl, the question is probably better answered there. However, most http clients in python with have a method for getting the url. [Check out this stack quote on it](https://stackoverflow.com/questions/23238862/get-url-using-beautiful-soup) -> "If you are using requests to load your pages, then simply ask it for the URL. [response.url](https://requests.readthedocs.io/en/latest/api/#requests.Response.url) tells you what URL the response was loaded from."

**Reply by Baboomba** [reply]
This is the part of my codes. I wrote it in python language.

But, the information I targeted at couldn't have been caught... It's the 'shepOwnerNumber' marked in bold below.

This site doesn't open API. So, I can't get this information.

The only way I can do is to check the Chrome Devtools' network console.
![picture](https://user-images.githubusercontent.com/116855194/200154946-c33f97e8-554c-4c41-b0b1-17b12d55e59f.JPG)







     while j_obj['contents'] != []:
            
                URL3 = r'https://self.baemin.com/v1/orders?orderStatus=CLOSED&offset={}0&limit=\
                    10&startDate=2022-10-01&endDate=2022-10-31&shopNumbers=&**shopOwnerNumber**={}'.format(k, shop_no[i])
                driver.get(URL3)
                p_source = driver.page_source
                soup = BeautifulSoup(p_source, "html.parser")
                parcing = soup.select_one('pre').text
                j_obj = json.loads(parcing)
                                
                for l in range(len(j_obj['contents'])):
                    goods_quantity = j_obj['contents'][l]['order']['items']
                    for m in range(len(goods_quantity)):
                        extract = goods_quantity[m]
                        q_list = list(extract.values())[:-2]
                        quantity.append(q_list)
                                
                k = k + 1
                time.sleep(1)
            
                df_q = pd.DataFrame(quantity)
                df_q.to_excel(r'menu_quantity_{}.xlsx'.format(i))

**Reply by david-engelmann** [reply]
Is this selenium?

**Reply by Baboomba** [reply]
yes, I wrote it in selenium!

**Comment by airtower-luna** [comment]
The thing marked in your screenshot is a *request* URL, not part of the response. So you'll have to look at what caused that request to be sent: Did you click at a link? Is there some script that creates those requests? I don't know if whatever caused the request has a list, but that's where you'd need to look.

**Reply by Baboomba** [reply]
![picture](https://user-images.githubusercontent.com/116855194/200345313-29f2c287-d953-43e7-a89e-f9b24ceb162d.JPG)
Yes, it is. The problem is that URL wasn't changed, if I've had clicked some buttons. But, I want to know the 'request URL'. How can I get, or crawl this information. I think it must be done, for Chrome Devtools has shown it.

Targeted information is 'shopOwnerNumber' in the request URL.

I can only check it in the DevTools' network tab like the image. I doesn't appear in URL.

Is there some way to get it?

**Reply by airtower-luna** [reply]
I don't know. I can see in your new screenshot that the "orders" request was triggered by code in a `venders.js` script. You'll have to look into that script and see how it builds the request. The dev tools might be able to point you at the line where the request is sent, see how it's built before that line.

**Reply by Baboomba** [reply]
OK, I'll study it more. Your advice help me so much.

Thanks😀

**Comment by david-engelmann** [comment, marked as answer]
where do you declare j_obj? Selenium has access to the current url the driver object is using with `driver.current_url`, [check it in the documentation](https://selenium-python.readthedocs.io/api.html?highlight=current_url#selenium.webdriver.remote.webdriver.WebDriver.current_url)!

**Reply by Baboomba** [reply]
This doc. is so good!

I'll study it more, thank you!

**Reply by david-engelmann** [reply]
@Baboomba Don't forget to "Mark as Answer"

**Comment by ghostinhershell** [comment]
Hey @Baboomba 👋 

If any of the responses here answered your question, please be sure to mark that response as the answer to your post so that anyone else that may have a similar question can reference this in the future. Thanks!

**Reply by Baboomba** [reply]
I'm sorry, I just did it!
